### DISCUSSION TITLE
V2V and V2I communication

### DISCUSSION THREAD

**Discussion by xrispa-m** [riebl/artery#199] [Q&A]
Just a quick question @riebl for something I have not fully understand. 
Is the CaService used for all the messages that the vehicles send to other vehicles and to any RSUs in range and the RsuCaService is for the messages sent from the RSUs? Or the CaService includes the V2V communication and RsuCaService includes the V2I communication? 
Are the messages broadcasted and any available OBU/RSU in range could receive them?

**Comment by riebl** [comment]
`CaService` runs on vehicles, `RsuCaService` runs on RSUs. Both services are generating CAMs but they include different container types, i.e. the CAM payload is different. There is no distinction between V2V or V2I: Vehicles with a `CaService` receive all CAMs, from vehicles and RSUs; the same applies to RSUs running a `RsuCaService`.
